IPython/core/async_helpers.py

```diff
@@ -0,0 +1,82 @@
+"""
+Async helper function that are invalid syntax on Python 3.5 and below.
+
+Known limitation and possible improvement. 
+
+Top level code that contain a return statement (instead of, or in addition to
+await) will be detected as requiring being wrapped in async calls. This should
+be prevented as early return will not work.
+"""
+
+
+
+import ast
+import sys
+import inspect
+from textwrap import dedent, indent
+from types import CodeType
+
+def _asyncio_runner(coro):
+    """
+    Handler for asyncio autoawait
+    """
+    import asyncio
+    return asyncio.get_event_loop().run_until_complete(coro)
```

Comment:
What would happen if ipykernel started running on asyncio? Would this cause hangs?